lib/solarwinds_apm/opentelemetry/otlp_processor.rb

```diff
@@ -88,17 +82,33 @@ def meter_attributes(span)
           meter_attrs['http.method'] = span.attributes[HTTP_METHOD] if span.attributes[HTTP_METHOD]
         end
         SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] meter_attrs: #{meter_attrs.inspect}" }
+        meter_attrs.compact!
         meter_attrs
       end
 
-      def calculate_lambda_transaction_name(span)
-        (ENV['SW_APM_TRANSACTION_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || span.name || 'unknown').slice(0, 255)
+      def calculate_lambda_transaction_name(span_name)
+        (ENV['SW_APM_TRANSACTION_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || span_name || 'unknown').slice(0, 255)
       end
 
-      def init_metrics
-        {
-          response_time: @meters['sw.apm.request.metrics'].create_histogram('trace.service.response_time', unit: 'ms', description: 'Duration of each entry span for the service, typically meaning the time taken to process an inbound request.')
-        }
+      # Get trans_name and url_tran of this span instance.
+      # Predecessor order: custom SDK > env var SW_APM_TRANSACTION_NAME > automatic naming
+      def calculate_transaction_names(span)
+        return calculate_lambda_transaction_name(span.name) if SolarWindsAPM::Utils.determine_lambda
+
+        trace_span_id = "#{span.context.hex_trace_id}-#{span.context.hex_span_id}"
+        trans_name = @txn_manager.get(trace_span_id)
+        if trans_name
+          SolarWindsAPM.logger.debug do
+            "[#{self.class}/#{__method__}] found trans name from txn_manager: #{trans_name} by #{trace_span_id}"
+          end
+          @txn_manager.del(trace_span_id)
+        elsif !ENV['SW_APM_TRANSACTION_NAME'].to_s.empty?
+          trans_name = ENV.fetch('SW_APM_TRANSACTION_NAME', nil)
+        else
+          trans_name = span.attributes[HTTP_ROUTE] || nil
+          trans_name = span.name if trans_name.to_s.empty? && span.name
+        end
```

Comment:
The truncation to 256 chars should be done for "non-lambda" transaction names too.

Reply:
Updated

Reply:
How about moving the truncation to outside of the else clause so it covers everything not just automatic route/span derived names?

Reply:
Updated.